Lombiq.Tests.UI/SecurityScanning/AutomationFrameworkPlans/Baseline.yml

```diff
@@ -42,8 +42,14 @@ jobs:
     enableTags: false
     disableAllRules: false
   rules:
-  - id: 10035
-    name: "Strict-Transport-Security Header"
+  # Some stock Orchard Core shapes and pages (including the setup page) contain inline script blocks, so this directive
+  # must be permitted for OC to work at all.
+  - id: 10055
+    name: "script-src includes unsafe-inline"
+    threshold: "off"
```

Comment:
On the setup screen, for sure. Now that I think about it, is there any way to identify that OC is in setup? because if so, we could simply not do inject the content security policy header until that has concluded.

Reply:
I guess you can check for `SetupController` in the `HttpContext`'s routes, as well as `ShellSettings.TenantState` should be something else than `Running`. And yes if this breaks the setup screen than rather the header shouldn't be injected.

Reply:
I managed to do that using `ShellSettings`. Unfortunately that only let me get rid of the `script-src unsafe-eval` rule.

We still need `script-src unsafe-inline`, because both Lombiq's and OC's codebase is absolutely littered with the stuff. After all, `script-src unsafe-inline` is needed for any script block, so if you do a global search for `<script at="Foot"`, there are 185 results in OC and 43 in OSOCE. All of those will have to be refactored into JS files before this rule can be re-enabled. We had some discussion about this in the past on Teams, I'm not convinced this is the way to go.

An alternative is to add a [cryptographic nonce](https://en.wikipedia.org/wiki/Cryptographic_nonce) (not to be confused with a [British nonce](https://en.wikipedia.org/wiki/Glossary_of_British_terms_not_widely_used_in_the_United_States#N:~:text=to%20his%20team.%22-,nonce,-a%20slang%20term), seriously wtf) to each script block that's delivered to that page. It could be done in Razor by extending the `ScriptTagHelper` to handle the `nonce` attribute. There is already an open issue in OC (https://github.com/OrchardCMS/OrchardCore/issues/13389) so I added a comment explaining my idea regarding the tag helper. For the time being we can't do anything else.

Also at least for TDEAL we still need `style-src unsafe-eval` because of [Slick](https://kenwheeler.github.io/slick/). It uses jQuery to add `style` attributes which violates the directive. This is a known problem, here is the open issue: https://github.com/kenwheeler/slick/issues/3799 We may replace Slick with a different carousel library in the future, but let's not scope creep in this issue.

Reply:
I see. The nonce would definitely be the desirable solution, I added that issue to our tracked issues. Then use the OC-compatible header with a comment linking the OC issue, please, and link in the YMLs too.

We don't speak Bri'ish here.

![image](https://github.com/Lombiq/UI-Testing-Toolbox/assets/1976647/ba3a2745-efe5-411b-a13c-bb95f3a7f350)